src/app/upgrade.cpp

```diff
@@ -110,13 +111,130 @@ namespace
             settingsStorage->removeValue(oldKey);
         }
     }
+
+    void upgradeSchedulerDaysSettings()
+    {
+        auto *settingsStorage = SettingsStorage::instance();
+        const auto key = QString::fromLatin1("Preferences/Scheduler/days");
+        const auto value = settingsStorage->loadValue<QString>(key);
+
+        bool ok = false;
+        const auto number = value.toInt(&ok);
+
+        if (ok)
+        {
+            switch (number)
+            {
+            case 0:
+                settingsStorage->storeValue(key, Scheduler::Days::EveryDay);
+                break;
+            case 1:
+                settingsStorage->storeValue(key, Scheduler::Days::Weekday);
+                break;
+            case 2:
+                settingsStorage->storeValue(key, Scheduler::Days::Weekend);
+                break;
+            case 3:
+                settingsStorage->storeValue(key, Scheduler::Days::Monday);
+                break;
+            case 4:
+                settingsStorage->storeValue(key, Scheduler::Days::Tuesday);
+                break;
+            case 5:
+                settingsStorage->storeValue(key, Scheduler::Days::Wednesday);
+                break;
+            case 6:
+                settingsStorage->storeValue(key, Scheduler::Days::Thursday);
+                break;
+            case 7:
+                settingsStorage->storeValue(key, Scheduler::Days::Friday);
+                break;
+            case 8:
+                settingsStorage->storeValue(key, Scheduler::Days::Saturday);
+                break;
+            case 9:
+                settingsStorage->storeValue(key, Scheduler::Days::Sunday);
+                break;
+            default:
+                LogMsg(QObject::tr("Invalid value found in configuration file, reverting it to default. Key: \"%1\". Value: \"%2\".")
+                    .arg(key, number), Log::WARNING);
```

Comment:
Is it correct? Shouldn't you have separate `arg()` for parameters of different types?

Reply:
Fixed.